iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/pipeconsensus/PipeConsensusAsyncConnector.java

```diff
@@ -0,0 +1,585 @@
+        tCommitId =
+            new TCommitId(
+                pipeRawTabletInsertionEvent.getCommitId(),
+                pipeRawTabletInsertionEvent.getRebootTimes());
+
+        // We increase the reference count for this event to determine if the event may be released.
+        if (!pipeRawTabletInsertionEvent.increaseReferenceCount(
+            PipeConsensusAsyncConnector.class.getName())) {
+          pipeRawTabletInsertionEvent.decreaseReferenceCount(
+              PipeConsensusAsyncConnector.class.getName(), false);
+          return;
+        }
+
+        final PipeConsensusTabletRawReq pipeConsensusTabletRawReq =
+            PipeConsensusTabletRawReq.toTPipeConsensusTransferReq(
+                pipeRawTabletInsertionEvent.convertToTablet(),
+                pipeRawTabletInsertionEvent.isAligned(),
+                tCommitId,
+                tConsensusGroupId,
+                thisDataNodeId);
+        final PipeConsensusTabletRawEventHandler pipeConsensusTabletRawEventHandler =
+            new PipeConsensusTabletRawEventHandler(
+                pipeRawTabletInsertionEvent, pipeConsensusTabletRawReq, this);
+
+        transfer(pipeConsensusTabletRawEventHandler);
+      }
+    }
+  }
+
+  private void transfer(
+      final PipeConsensusTabletBatchEventHandler pipeConsensusTabletBatchEventHandler) {
+    AsyncPipeConsensusServiceClient client = null;
+    try {
+      client = asyncTransferClientManager.borrowClient(getFollowerUrl());
+      pipeConsensusTabletBatchEventHandler.transfer(client);
+    } catch (final Exception ex) {
+      logOnClientException(client, ex);
+      pipeConsensusTabletBatchEventHandler.onError(ex);
+    }
+  }
+
+  private void transfer(
+      final PipeConsensusTabletInsertNodeEventHandler pipeConsensusInsertNodeReqHandler) {
+    AsyncPipeConsensusServiceClient client = null;
+    try {
+      client = asyncTransferClientManager.borrowClient(getFollowerUrl());
+      pipeConsensusInsertNodeReqHandler.transfer(client);
+    } catch (final Exception ex) {
+      logOnClientException(client, ex);
+      pipeConsensusInsertNodeReqHandler.onError(ex);
+    }
+  }
+
+  private void transfer(final PipeConsensusTabletRawEventHandler pipeConsensusTabletReqHandler) {
+    AsyncPipeConsensusServiceClient client = null;
+    try {
+      client = asyncTransferClientManager.borrowClient(getFollowerUrl());
+      pipeConsensusTabletReqHandler.transfer(client);
+    } catch (final Exception ex) {
+      logOnClientException(client, ex);
+      pipeConsensusTabletReqHandler.onError(ex);
+    }
+  }
+
+  @Override
+  public void transfer(TsFileInsertionEvent tsFileInsertionEvent) throws Exception {
+    boolean enqueueResult = addEvent2Buffer(tsFileInsertionEvent);
+    if (!enqueueResult) {
+      throw new PipeException(ENQUEUE_EXCEPTION_MSG);
+    }
+
+    syncTransferQueuedEventsIfNecessary();
+    transferBatchedEventsIfNecessary();
+
+    if (!(tsFileInsertionEvent instanceof PipeTsFileInsertionEvent)) {
```

Comment:
"Enqueue" after the judgement, can be after "increaseReferenceCount".

Reply:
ditto